Clases/Ataque.cpp

```diff
@@ -16,18 +17,22 @@ Ataque::Ataque(sf::Vector2f pos, bool left, const sf::Texture &tx) {
 bool Ataque::dibujar(sf::RenderWindow &w, LinkedList<sf::Rect<float> *> list, LinkedList<Enemigos*> enemy) {
     w.draw(sp);
     sf::Rect<float> *miRectangulo;
-    //sf::Vector2f *enemigo;
+    sf::Rect<float> enemigo;
 
-    for(int ii = 0; ii < list.getSize(); ii++){
-        miRectangulo = list.get(ii);
-        //enemigo = enemy.get(ii);
+    for(list.iterInit(); !list.iterEnd(); list.iterNext()){
```

Comment:
Cambie el for de ii al for con el "iterador" de la lista. Es bastante mas eficiente